### PR TITLE
Variable storage

### PR DESCRIPTION
**Pull request by Paul-Ferrell**
- Deferred are more flexible
   - They can contain multiple values
   - They don't need to be named, or have their elements pre-defined.
   - Variables that reference other variables become deferred if
     they reference something deferred.
 - Each test run is handed its variable manager now
   - The variables are saved and loaded when a run object is loaded.
   - Provided mechanisms for updating the remaining deferred vars.
   - Provided mechanisms for updating the test run config too.
   - Added a finalize method to do the two prior things in one shot.
 - The run.tmpl script isn't really a template any more, just a trial run.
   - run.sh is now written completely from scratch after deferred
     variables are resolved.
 - We can now add any pavilion variable value directly to
   the result defaults.
   - Added sys_name, user, and job_id (which isn't a var, but still).
- Output functions (fprint, etc) are now in output.py.

Compatability Breaking:
 - Changed the argument for SystemVariables from plugin_name to name

### CODE REVIEW

**Comment by Paul-Ferrell** [comment]
Fixes #70

**Comment by Paul-Ferrell** [comment]
Fixes #67

File: test/tests/test_setup_tests.py [hunk 1]
```diff
@@ -287,6 +288,43 @@ def test_for_circular_variable_references(self):
...
+        fin_var_man.add_var_set('sys', undefered_sys_vars)
+
+        test.finalize(fin_var_man)
+
+        results = test.gather_results(test.run())
+        test.save_results(results)
+
+        plugins._reset_plugins()
+
     def test_resolve_all_vars(self):
```
Comment (nicholas-sly) [inline test/tests/test_setup_tests.py]: Consider updating the test name to match the new function name (resolve_config).

File: test/tests/wait_cmd_tests.py [hunk 2]
```diff
@@ -1,9 +1,7 @@
 from pavilion import commands
 from pavilion import plugins
-from pavilion import status_file
-from pavilion import system_variables
 from pavilion.series import TestSeries
-from pavilion.test_config import file_format
+from pavilion.test_config import file_format, VariableSetManager
 from pavilion.unittest import PavTestCase
 from pavilion.test_run import TestRun
```
Comment (nicholas-sly) [inline test/tests/wait_cmd_tests.py]: Bump over unittest.

File: bin/pav-lib.bash [hunk 3]
```diff
@@ -316,20 +316,20 @@ function verify_module_loaded() {
         1)
             msg="Module ${module_name}, version ${module_version} was not "
             msg="${msg}loaded. See the test log."
-            ${PAV_PATH} set_status -s ENV_FAILED -n "${msg}" ${test_id}
+            ${PAV_PATH} set_status -s ENV_FAILED -n"${msg}" ${test_id}
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply bin/pav-lib.bash]: There's a ticket on the subject. On cray, for some reason, this doesn't work with the space.

**Pull request merged**